biocrnpyler/core/component.py

```diff
@@ -63,6 +65,11 @@ def __init__(self, name: Union[str, Species],
         else:
             self.set_mixture(None)
 
+        if compartment is not None:
+            if not isinstance(compartment, Compartment):
+                raise TypeError("The compartment must be a Compartment object")
+            self.compartment = compartment
+     
```

Comment:
In other places (like `Species`), we allow the compartment to be a string.  Shouldn't that be allowed here as well (and then it gets converted internally to a `Compartment` object?).

Reply:
Usually, a component out of the core/components/ library will create the Component class and should handle the compartment creation for the component (like IntegralMembraneProtein handles its compartments and allows strings).